ls1intum/Artemis
# `Exam mode`: Allow instructors to enable the test download for students

### Checklist
#### General


#### Client

### Motivation and Context


#5762 allowed students to download the tests after the due date if the instructor enables it. This should be available for exam exercises as well.

### Description

Instructors can now enable the download of the tests for exam exercises as well after setting a publication date in the exam itself.

### Steps for Testing

Prerequisites:
- 1 Instructor
- 1 Student

1. Create an exam without a date for the release of the example solutions and with a programming exercise
2. Participate as student
3. Check that you cannot enable including the tests in the edit screen of the exercise as instructor
4. If the due date of the exam is not over yet, wait until it is
5. Set a publication date for the example solution in the exam in the past
6. Check that the student can now download the example solution
7. Enable the inclusion of the tests in the programming exercise
8. Check that the student can now download the tests as well

### Testserver States
> [!NOTE]
> These badges show the state of the test servers.
> Green = Currently available, Red = Currently locked

![](https://byob.yarr.is/ls1intum/Artemis/artemis-test1)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test2)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test3)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test4)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test5)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test6)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test9)

### Review Progress






#### Performance Review
#### Code Review
#### Manual Tests

### Test Coverage





#### Client

| Class/File | Line Coverage | Confirmation (assert/expect) |
|------------|--------------:|---------------------:|
| [exam-result-summary.component.ts](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5777/latest/artifact/shared/Coverage-Report-Client-Tests/app/exam/participate/summary/exam-result-summary.component.ts.html) | 91.35% | ✅ |

### Screenshots


![Bildschirmfoto 2023-11-23 um 13 58 29](https://github.com/ls1intum/Artemis/assets/38322605/ac116ec1-0815-4749-b1fe-3c04ecd42093)
